torch/nn/modules/unfold.py

```diff
@@ -0,0 +1,67 @@
+import torch
+from torch.autograd import Variable
+
+from .module import Module
+from .utils import _single, _pair, _triple
+from .._functions.thnn.unfold import Im2Col
+
+
+class Unfold(Module):
+    """
+    Rearranges image blocks into columns.
+
+    Converts each sliding :math:`kernel_size` block of the input into a column of
+    the output.  The output has :math:`\prod_i k_i, k_i\in kernel_size` rows and
+    contains as many columns as there are :math:`kernel_size` neighborhoods of
+    the input according to the stride and dilation values.
+
+    | If :attr:`padding` is non-zero, then the input is implicitly
+    zero-padded on both sides by :attr:`padding` number of points
+    | :attr:`dilation` controls the spacing between the kernel points.
+    It is harder to describe, but this `link`_ has a nice visualization of what
+    dilation does.
+
+    Args:
+        kernel_size (int or tuple): the size of the neighborhoods to convert
+        stride (int or tuple): the stride of the sliding neighborhood.
+                               Default: 1
+        padding (int or tuple, optional): implicit zero padding to be added on
+                                          both sides of input. Default: 0
+        dilation (int or tuple, optional): a parameter that controls the
+                                           vertical stride of elements in the
+                                           neighborhood
+
+    Shape:
+        - Input: :math:`(C, L_{in})`
+        - Output: :math:`(C * \prod(kernel_size), L_{out},)` where
```

Comment:
I'm not too familiar with im2col and what the use cases are. Is the output supposed to be (C * \prod(kernel_size), L_{out}) and not something like (C, \prod(kernel_size), L_{out})?